src/com/facebook/buck/swift/SwiftPrepareForCompile.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.swift;
+
+import com.facebook.buck.cxx.CxxPrepareForLinkStep;
+import com.facebook.buck.io.ProjectFilesystem;
+import com.facebook.buck.model.BuildTargets;
+import com.facebook.buck.rules.AbstractBuildRule;
+import com.facebook.buck.rules.AddToRuleKey;
+import com.facebook.buck.rules.BuildContext;
+import com.facebook.buck.rules.BuildRuleParams;
+import com.facebook.buck.rules.BuildableContext;
+import com.facebook.buck.rules.SourcePath;
+import com.facebook.buck.rules.SourcePathResolver;
+import com.facebook.buck.rules.args.Arg;
+import com.facebook.buck.rules.args.FileListableLinkerInputArg;
+import com.facebook.buck.rules.args.SourcePathArg;
+import com.facebook.buck.rules.args.StringArg;
+import com.facebook.buck.step.Step;
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import java.nio.file.Path;
+
+import javax.annotation.Nullable;
+
+class SwiftPrepareForCompile extends AbstractBuildRule {
+
+  @AddToRuleKey(stringify = true)
+  private final Path outputPath;
+
+  @AddToRuleKey
+  private final ImmutableSet<SourcePath> srcs;
+
+  private final Path fileListPath;
+
+  SwiftPrepareForCompile(
+      BuildRuleParams buildRuleParams,
+      SourcePathResolver resolver,
+      Path outputPath,
+      ImmutableSet<SourcePath> srcs) {
+    super(buildRuleParams, resolver);
+    this.outputPath = outputPath;
+    this.srcs = srcs;
+
+    fileListPath = getProjectFilesystem().resolve(
+        BuildTargets.getScratchPath(getProjectFilesystem(), getBuildTarget(), "%s__filelist.txt"));
+  }
+
+  private Step createFileListStep() {
+    ImmutableList<Arg> sourceListArgs = FileListableLinkerInputArg.from(
+        FluentIterable.from(SourcePathArg.from(getResolver(), srcs))
+            .filter(SourcePathArg.class)
+            .toList());
+
+    ProjectFilesystem fileSystem = getProjectFilesystem();
+    Path argFilePath = fileSystem.getRootPath().resolve(
```

Comment:
we actually don't need argfile for swift, but I don't want to duplicate stuff by creating similar class like `CxxPrepareForLinkStep`.